src/Context.ts

```diff
@@ -87,16 +97,14 @@ class Context implements IContext {
   }
 }
 
-function getLogger(invocationId: string, functionName: string, log: ILogCallback): ILogger{
-    return Object.assign(
-      <ILog>(...args: any[]) => log(LogLevel.Information, ...args),
-      {
-        error: <ILog>(...args: any[]) => log(LogLevel.Error, ...args),
-        warn: <ILog>(...args: any[]) => log(LogLevel.Warning, ...args),
-        info: <ILog>(...args: any[]) => log(LogLevel.Information, ...args),
-        verbose: <ILog>(...args: any[]) => log(LogLevel.Trace, ...args)
-      }
-    );
+// Emit warning if trying to log after function execution is done.
+function logWithAsyncCheck(done: boolean, log: ILogCallback, level: LogLevel, ...args: any[]) {
+  if (done) {
+    let badAsyncMsg = "Error: Unexpected call to 'log' after function execution has completed. Please check for asynchronous calls that are not awaited or did not use the 'done' callback where expected.";
+    log(LogLevel.Error, badAsyncMsg);

```

Comment:
Should we make this a Warning instead of an Error? I worry about existing customers that are happily running with this problem today (on accident) suddenly getting a huge wave of errors when we roll this out. Warnings would be a little more acceptable, I think.

Reply:
Agreed.

The only other comment I have is a small nit on the test function name for the promise+callback scenario, which is currently `asyncThrowsError`, something more descriptive would be helpful.